x/bep3/types/params.go

```diff
@@ -0,0 +1,134 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+)
+
+// Parameter keys
+var (
+	KeyBnbDeputyAddress = []byte("BnbDeputyAddress")
+	KeyMinBlockLock     = []byte("MinBlockLock")
+	KeyMaxBlockLock     = []byte("MaxBlockLock")
+	KeySupportedAssets  = []byte("SupportedAssets")
+
+	AbsoluteMaximumBlockLock int64          = 10000
+	AbsoluteMinimumBlockLock int64          = 10
+	DefaultBnbDeputyAddress  sdk.AccAddress = sdk.AccAddress("kava1xy7hrjy9r0algz9w3gzm8u6mrpq97kwta747gj")
+	DefaultMinBlockLock      int64          = 20
+	DefaultMaxBlockLock      int64          = 200
+	DefaultSupportedAssets                  = AssetParams{AssetParam{Denom: "kava", CoinID: "459", Limit: 1, Active: false}}
+)
+
+// Params governance parameters for bep3 module
+type Params struct {
+	BnbDeputyAddress sdk.AccAddress `json:"bnb_deputy_address" yaml:"bnb_deputy_address"` // Bnbchain deputy address
+	MinBlockLock     int64          `json:"min_block_lock" yaml:"min_block_lock"`         // AtomicSwap minimum block lock
+	MaxBlockLock     int64          `json:"max_block_lock" yaml:"max_block_lock"`         // AtomicSwap maximum block lock
+	SupportedAssets  AssetParams    `json:"supported_assets" yaml:"supported_assets"`     // supported assets
+}
+
+// String implements fmt.Stringer
+func (p Params) String() string {
+	return fmt.Sprintf(`Params:
+	Bnbchain deputy address: %s,
+	Min block lock: %d,
+	Max block lock: %d,
+	Supported assets: %s`,
+		p.BnbDeputyAddress, p.MinBlockLock, p.MaxBlockLock, p.SupportedAssets)
+}
+
+// NewParams returns a new params object
+func NewParams(bnbDeputyAddress sdk.AccAddress, minBlockLock int64, maxBlockLock int64, supportedAssets AssetParams) Params {
+	return Params{
+		BnbDeputyAddress: bnbDeputyAddress,
+		MinBlockLock:     minBlockLock,
+		MaxBlockLock:     maxBlockLock,
+		SupportedAssets:  supportedAssets,
+	}
+}
+
+// DefaultParams returns default params for bep3 module
+func DefaultParams() Params {
+	return NewParams(DefaultBnbDeputyAddress, DefaultMinBlockLock, DefaultMaxBlockLock, DefaultSupportedAssets)
+}
+
+// AssetParam governance parameters for each asset within a supported chain
+type AssetParam struct {
+	Denom  string `json:"denom" yaml:"denom"`     // name of the asster
+	CoinID string `json:"coin_id" yaml:"coin_id"` // internationally recognized coin ID
+	Limit  int64  `json:"limit" yaml:"limit"`     // asset supply limit
```

Comment:
nit: I would use `sdk.Int`

Reply:
Addressed in https://github.com/Kava-Labs/kava/pull/370/commits/f73778e91076600eed68817474ce2e1e92593daa.